nuget/Microsoft.Windows.CsWinRT.targets

```diff
@@ -316,6 +322,242 @@ $(CsWinRTInternalProjection)
+    <PropertyGroup>
+    
+      <!-- Prepare the generated files directory -->
+      <_StubExeFolderName>StubExe</_StubExeFolderName>
+      <_StubExeGeneratedFilesDir Condition="'$(_StubExeGeneratedFilesDir)' == '' AND '$(GeneratedFilesDir)' != ''">$(GeneratedFilesDir)\$(_StubExeFolderName)\</_StubExeGeneratedFilesDir>
+      <_StubExeGeneratedFilesDir Condition="'$(_StubExeGeneratedFilesDir)' == ''">$([MSBuild]::NormalizeDirectory('$(MSBuildProjectDirectory)', '$(IntermediateOutputPath)', 'Generated Files', '$(_StubExeFolderName)'))</_StubExeGeneratedFilesDir>
+    
+      <!-- The path of the .lib file produced by the Native AOT toolchain, which we need for the linker -->
+      <_StubExeNativeLibraryPath>$([System.IO.Path]::Combine($(MSBuildProjectDirectory), $(NativeOutputPath), $(AssemblyName).lib))</_StubExeNativeLibraryPath>
+    
+      <!-- Prepare the path of the .c file with the stub .exe source we need to compile -->
+      <_StubExeSourceFileName>$(AssemblyName).c</_StubExeSourceFileName>
+      <_StubExeSourceFilePath>$([System.IO.Path]::Combine($(_StubExeGeneratedFilesDir), $(_StubExeSourceFileName)))</_StubExeSourceFilePath>
+    
+      <!-- Prepare the output path of the resulting stub .exe -->
+      <_StubExeBinaryFileName>$(AssemblyName).exe</_StubExeBinaryFileName>
+      <_StubExeBinaryOutputFilePath>$([System.IO.Path]::Combine($(_StubExeGeneratedFilesDir), $(_StubExeBinaryFileName)))</_StubExeBinaryOutputFilePath>
+      <_StubExeBinaryDestinationFilePath>$([System.IO.Path]::Combine($(MSBuildProjectDirectory), $(NativeOutputPath), $(_StubExeBinaryFileName)))</_StubExeBinaryDestinationFilePath>
+
+      <!-- Get the target platform (we either use 'Platform', if set and not 'AnyCPU', or fallback to 'PlatformTarget' ) -->
+      <_StubExePlatform Condition="'$(_StubExePlatform)' == '' AND '$(Platform)' != '' AND '$(Platform)' != 'AnyCPU'">$(Platform)</_StubExePlatform>
+      <_StubExePlatform Condition="'$(_StubExePlatform)' == '' AND '$(PlatformTarget)' != ''">$(PlatformTarget)</_StubExePlatform>
+    </PropertyGroup>
+    
+    <!--
+      This is a simple C source file just jumping into the special 'Main' program of the current project.
+      Note that we need to use '%3B' to escape the semicolons, as otherwise it'll be skipped entirely.
+    -->
+    <PropertyGroup>
+      <_StubExeSourceFileLines>
+        <![CDATA[#include <wchar.h>
+
+// Declare the import for '__managed__Main', which is a special export produced by the
+// Native AOT runtime to allow jumping into the real 'Main' of an application. This is
+// exported automatically when the 'CustomNativeMain' property is set to 'true'.
+__declspec(dllimport)
+int __stdcall __managed__Main(int argc, wchar_t** argv)%3B
+
+// Our entry point is simply a direct jump into the entry point from Native AOT
+int wmain(int argc, wchar_t** argv)
+{
+    return __managed__Main(argc, argv)%3B
+}]]>
+      </_StubExeSourceFileLines>
```

Comment:
Could the CsWinRT package include the stub C file instead of embedding the source in MSBuild?

Reply:
Yeah we can do that! I thought we'd need to modify it but now that it's always the same it's fine 🙂